src/datasources/relay-api/gelato-api.service.ts

```diff
@@ -63,4 +73,37 @@ export class GelatoApi implements IRelayApi {
   private getRelayGasLimit(gasLimit: bigint): bigint {
     return gasLimit + GelatoApi.GAS_LIMIT_BUFFER;
   }
+
+  async getRelayCount(args: {
+    chainId: string;
+    address: string;
+  }): Promise<number> {
+    const cacheDir = this.getRelayCacheKey(args);
+    const count = await this.cacheService.get(cacheDir);
+    return count ? parseInt(count) : 0;
+  }
+
+  async setRelayCount(args: {
+    chainId: string;
+    address: string;
+    count: number;
+  }): Promise<void> {
+    const cacheDir = this.getRelayCacheKey(args);
+    await this.cacheService.set(
+      cacheDir,
+      args.count.toString(),
+      this.ttlSeconds,
+    );
+  }
```

Comment:
This is out of the scope of this PR, but maybe at some point it would be nice to refactor `RelayRepository.relay` to use an atomic operation for getting and incrementing the counter value, instead of separate operations. (We can leverage the already existing `ICacheService.increment` function, which server this purpose).

Although I think it's not critical, as the case where the counter can be retrieved from the cache but at the same time the increased value cannot be persisted shouldn't happen, it would be the best practice IMO.

Reply:
I agree, and that it should be done separately. I will make a note to open a follow up PR with the refactor.